system/jlib/jptree.cpp

```diff
@@ -3046,6 +3045,56 @@ void PTree::serializeToStream(IBufferedSerialOutputStream &tgt) const
     serializeCutOff(tgt, -1, 0);
 }
 
+void PTree::deserializeFromStream(IBufferedSerialInputStream &src)
+{
+    deserializeSelf(src);
+
+    StringBuffer eName;
+    size32_t len{0};
+    for (;;)
+    {
+        const char *nextName = queryZeroTerminatedString(src, len);
+        if (len == 0)
```

Comment:
I don't think is quite right. If there were no chars in stream queryZeroTerminatedString would return null and len would be left as 0.

This condition need to ensure that it has actually read a null string, if it is len would be 1, 0 is an error.